lib/services/messenger/translators/discourse.ts

```diff
@@ -0,0 +1,257 @@
+				instance: this.connectionDetails.instance,
+			};
+		});
+	}
+
+	public messageIntoEmitDetails(message: TransmitInformation): Promise<DiscourseEmitInstructions> {
+		const title = message.details.title;
+		const thread = message.target.thread;
+		switch (message.action) {
+			case MessageAction.CreateThread:
+				if (!title) {
+					return Promise.reject(
+						new ProcBotError(ProcBotErrorCode.ServiceSpecificError, 'Cannot create a thread without a title.')
+					);
+				}
+				return Promise.resolve({ method: ['request'], payload: {
+					htmlVerb: 'POST',
+					path: '/posts',
+					body: {
+						category: message.target.flow,
+						raw: `${message.details.text}\n\n---\n${Translator.stringifyMetadata(message, 'logo')}`,
+						title,
+						unlist_topic: 'false',
+					},
+				}});
+			case MessageAction.CreateMessage:
+				if (!thread) {
+					return Promise.reject(
+						new ProcBotError(ProcBotErrorCode.ServiceSpecificError, 'Cannot create a comment without a thread.')
+					);
+				}
+				return Promise.resolve({ method: ['request'], payload: {
+					htmlVerb: 'POST',
+					path: '/posts',
+					body: {
+						raw: `${message.details.text}\n\n---\n${Translator.stringifyMetadata(message, 'logo')}`,
+						topic_id: thread,
+						whisper: message.details.hidden ? 'true' : 'false',
+					}
+				}});
+			case MessageAction.ReadConnection:
+				if (!thread) {
+					return Promise.reject(
+						new ProcBotError(ProcBotErrorCode.ServiceSpecificError, 'Cannot search for connections without a thread.')
+					);
+				}
+				return Promise.resolve({ method: ['request'], payload: {
+					htmlVerb: 'GET',
+					path: '/search/query',
+					qs: {
+						term: `Connects to [${message.source.service} thread`,
+						'search_context[type]': 'topic',
+						'search_context[id]': thread,
+					}
+				}});
+			case MessageAction.UpdateTags:
+				if (!thread) {
+					return Promise.reject(
+						new ProcBotError(ProcBotErrorCode.ServiceSpecificError, 'Cannot update tags without a thread.')
+					);
+				}
+				const tags = message.details.tags;
+				if (!_.isArray(tags)) {
+					return Promise.reject(
+						new ProcBotError(ProcBotErrorCode.ServiceSpecificError, 'Cannot update tags without a tags array.')
+					);
+				}
+				const getTopic = {
+					json: true,
+					method: 'GET',
+					qs: {
+						api_key: this.connectionDetails.token,
+						api_username: this.connectionDetails.username,
+					},
+					url: `https://${this.connectionDetails.instance}/t/${message.target.thread}`,
+				};
+				return request(getTopic)
+				.then((topicResponse) => {
+					return { method: ['request'], payload: {
+						body: {},
+						htmlVerb: 'PUT',
+						qs: {
+							'tags[]': tags,
+						},
+						path: `/t/${topicResponse.slug}/${thread}.json`,
+					}};
+				});
+			default:
+				return Promise.reject(new ProcBotError(
+					ProcBotErrorCode.ServiceSpecificError, `${message.action} not translatable to ${message.target.service} yet.`
```

Comment:
As this is a Messenger specific error, do we not need a MessengerError class which would be used in all these subclasses/interfaces? (Based on comment for `procbot.ts`). I'd expect the bot using the MessengerService (so SyncBot) to be checking the instances of the errors being returned to determine how to log/report them.
Applies to all 'ServiceSpecificError` instances.

Reply:
Note to self: Advice sought over desired error structure.